gwpy/timeseries/io/gwf/lalframe.py

```diff
@@ -155,6 +155,19 @@ def read(source, channels, start=None, end=None, series_class=TimeSeries,
     for name in channels:
         ts = _read_channel(stream, str(name), start=start, duration=duration)
         out[name] = series_class.from_lal(ts, copy=False)
+        tsend = ts.epoch + getattr(ts.data.data, "size", 0) * ts.deltaT
```

Comment:
I assume there is no getter for this length?

Reply:
No, because it's a SWIG-wrapped C struct, the access is fairly low-level.